src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -456,6 +502,28 @@ void ImageReplayer<I>::handle_init_remote_journaler(int r) {
     return;
   }
 
+  {
```

Comment:
I can't move it to `handle_bootstrap`, because the JournalMetadata will only start to listen for metadata updates after its initialization in method `init()`. Therefore, in order to not loose any resync request, I can only add the metadata listener after the initialization of the journaler.